File: sdk/dotnet/Mics.Client/MicsClientOptions.cs
namespace Mics.Client;

public sealed record MicsClientOptions(
    TimeSpan ConnectTimeout,
    TimeSpan AckTimeout,
    int MaxSendAttempts,
    TimeSpan HeartbeatInterval,
    bool AutoReconnect,
    TimeSpan ReconnectMinDelay,
    TimeSpan ReconnectMaxDelay,
    IMicsMessageCrypto? MessageCrypto)
{
    public static MicsClientOptions Default { get; } = new(
        ConnectTimeout: TimeSpan.FromSeconds(5),
        AckTimeout: TimeSpan.FromSeconds(3),
        MaxSendAttempts: 3,
        HeartbeatInterval: TimeSpan.FromSeconds(10),
        AutoReconnect: true,
        ReconnectMinDelay: TimeSpan.FromMilliseconds(200),
        ReconnectMaxDelay: TimeSpan.FromSeconds(5),
        MessageCrypto: null);
}

File: sdk/android/mics-client-sdk/mics-client-sdk/src/main/kotlin/com/mics/clientsdk/MicsClientOptions.kt
package com.mics.clientsdk

import java.time.Duration

data class MicsClientOptions(
    val connectTimeout: Duration = Duration.ofSeconds(5),
    val ackTimeout: Duration = Duration.ofSeconds(3),
    val maxSendAttempts: Int = 3,
    val heartbeatInterval: Duration = Duration.ofSeconds(10),
    val autoReconnect: Boolean = true,
    val reconnectMinDelay: Duration = Duration.ofMillis(200),
    val reconnectMaxDelay: Duration = Duration.ofSeconds(5),
    val messageCrypto: MessageCrypto? = null,
)

